File: summary/rust/11_prime_factorization.rs
fn prime_factors(mut n: u64) -> Vec<u64> {
    let mut factors = Vec::new();
    let mut divisor = 2;
    while n > 1 {
        while n % divisor == 0 {
            factors.push(divisor);
            n /= divisor;
        }
        divisor += 1;
    }
    factors
}

fn main() {
    let n = 60;
    println!("素因数分解結果: {:?}", prime_factors(n));
}

File: summary/c++/11_prime_factorization.cpp
#include <iostream>
#include <vector>

std::vector<int> prime_factors(int n) {
    std::vector<int> factors;
    for (int divisor = 2; n > 1; divisor++) {
        while (n % divisor == 0) {
            factors.push_back(divisor);
            n /= divisor;
        }
    }
    return factors;
}

int main() {
    int n;
    std::cout << "Input a number: ";
    std::cin >> n;
    std::vector<int> factors = prime_factors(n);
    std::cout << "Prime factors: ";
    for (size_t i = 0; i < factors.size(); i++) {
        std::cout << factors[i];
        if (i < factors.size() - 1) {
            std::cout << ", ";
        }
    }
    std::cout << std::endl;
    return 0;
}
